### PR TITLE
Add tooltips to the Suggestions Control

### PR DESCRIPTION
**Pull request by zadjii-msft**
_targets #14938_

These are currently powered by the `"ToolTip"` element in the completion protocol. This also adds room for `"description"`s in _any_ action. This will be more relevant for Tasks, in #1595. 

By all accounts, this could have been in the parent commit. I broke it apart for the ease of reviewing. 

* Related to #3212
* Doesn't close anything by itself

### CODE REVIEW

**Comment by zadjii-msft** [comment]
gh-3121-tooltips-for-pr

File: src/cascadia/TerminalApp/SuggestionsControl.xaml [hunk 1]
```diff
@@ -206,6 +206,17 @@
                       ItemsSource="{x:Bind FilteredActions}"
                       SelectionChanged="_listItemSelectionChanged"
                       SelectionMode="Single" />
+            <mux:TeachingTip x:Name="DescriptionTip"
+                             Title=""
+                             IsOpen="False"
+                             PreferredPlacement="Right"
+                             ShouldConstrainToRootBounds="False"
+                             Subtitle="">
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/SuggestionsControl.xaml:214]: Is "Subtitle" necessary? Why?

Comment (carlos-zamora) [reply src/cascadia/TerminalApp/SuggestionsControl.xaml:214]: Same with Title?

Also, looks like the WinUI docs don't say the default value (grr). So, mind checking each of these to make sure they're necessary?

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Accessibility is gonna be rough here. You should test it out, but I assume TeachingTips aren't read out by screen readers. What you could do though, is apply that text to the `AutomationProperties.FullDescription` property. That way, the user can specifically request more info whenever they want it. That info shouldn't be read out automatically by the screen reader either. Only on request.

**Comment by zadjii-msft** [comment]
> I assume TeachingTips aren't read out by screen readers

gods I forgot they aren't. WHY ARENT THEY. 

Also, Narrator and the Command Palette are already weird. Like, navigating the cmdpal with narrator never moves the selected item. It moves focus, but not the selected item. 

So of course, for the tooltips, they too do not get updated with narrator open. Frick.

**Comment by zadjii-msft** [comment]
Notes:

* [ ] `F6` to focus the TeachingTip doesn't work. It doesn't work because the `SuggestionsControl::_lostFocusHandler` fires, we don't find the tip as a child of the sxnui (it's a popup). We dismiss ourselves immediately.
  * Same with clicking on it.
  * This can be fixed by looking to see if the focused thing is a child of the 
* [ ] Closing the TeachingTip doesn't focus the sxnui again. The sxnui has lost focus (it lost focus to the popup), and the popup gave it to the TermControl, and now there's no way to dismiss the sxnui.
  * fixable. Toss focus manually. 
* [ ] OS dark theme, app in light theme? That's right, the `Title` of the TeachingTip will be white. Fucking themeing lambda again. 
* [ ] Change the app theme at runtime? sxnui doesn't reload it. 
  * [ ] I don't believe the cmdpal does either tho
* [ ] more?

I've got a vibe that this is too wacky to land for 1.19. I'm gonna close this one for now and come back to it. 

Further reading:
* https://github.com/microsoft/microsoft-ui-xaml/issues/3277#issuecomment-692369745
  * > However you should be able to adjust what narrator reads by setting automation properties on your TeachingTip. Setting AutomationID of the tip eventually gets folded into the announcement from narrator here

**Comment by zadjii-msft** [comment]
Attempted resurrection in `dev/migrie/fhl/sxnui-tooltips-3`

image

Less wacky with the TeachingTip, but positioning the menu gets harder and I haven't solved that yet.

**Comment by zadjii-msft** [comment]
AHAHAHA I finally got it to work right!

suggestions-tooltips-final
https://github.com/microsoft/terminal/compare/dev/migrie/f/sxnui-tooltips-4

**Pull request closed without merging**